iib/workers/tasks/build.py

```diff
@@ -496,6 +497,65 @@ def _opm_index_add(
 
     with set_registry_token(overwrite_from_index_token, from_index):
         run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to add the bundles to the index image')
+    
+    # add properties to bundles in the index
+    if add_properties:
+        maxOpenShiftVersionProperty = {
+            "type": "olm.maxOpenShiftVersion",
+            "value": "4.8",
+            "operatorbundle_version": "",
+            "operatorbundle-path": "",
+        }
+        for bundle in bundle_str.split(','):
+            if _sdk_is_valid(bundle):
+                _add_property_to_index(base_dir, bundle, maxOpenShiftVersionProperty)
+
+
+@retry(attempts=2, wait_on=IIBError, logger=log)
+def _add_property_to_index(base_dir, bundle, property):
```

Comment:
```suggestion
// Add the property max ocp version. This function is required for we are able to avoid workflow issues. 
// We need to ensure that any bundle which has deprecated/removed API(s) in 1.22/ocp 4.9 will be using
// the property to not allow the users to upgrade the cluster to 4.9 before upgrade the operator installed
// to a version that is compatible with 4.9(migrated)
@retry(attempts=2, wait_on=IIBError, logger=log)
def _add_property_to_index(base_dir, bundle, property):
```

Reply:
nit: I think would be better we add a comment and a name that clarifies better its intention.